crates/wasi-common/src/snapshots/wasi_snapshot_preview1.rs

```diff
@@ -643,17 +648,20 @@ impl<'a> WasiSnapshotPreview1 for WasiCtx {
         } else {
             types::Rights::FD_SEEK | types::Rights::FD_TELL
         };
-        let mut entry = self.get_entry_mut(fd)?;
+        let entry = self.get_entry(fd)?;
         let file = entry
-            .as_descriptor_mut(rights, types::Rights::empty())?
-            .as_file_mut()?;
+            .as_descriptor(rights, types::Rights::empty())?
+            .as_file()?;
         let pos = match whence {
             types::Whence::Cur => SeekFrom::Current(offset),
             types::Whence::End => SeekFrom::End(offset),
             types::Whence::Set => SeekFrom::Start(offset as u64),
         };
         let host_newoffset = match file {
-            Descriptor::OsHandle(fd) => fd.seek(pos)?,
+            Descriptor::OsHandle(fd) => {
+                let mut fd: &File = fd;
+                fd.seek(pos)?
```

Comment:
For one-liners like this it should suffice to do `(&fd).seek(pos)`

Reply:
I tried but alas wouldn’t work. The compiler complained that cannot mutate `File` behind a `&` ref. I think this might be since `fd` is not a `File` but rather an entity that implements `Deref<Target=File>` so would need something like `(&mut (&*fd as &File))`. Unless I’m doing something wrong here.

Reply:
Ok, I've convinced myself that this is due to the fact that `fd` is an `OsHandle` and not a `File` nor `&File`. Hence, we need to force it to coerce to `&File` before calling the trait method such as `seek` on it. The minimum working one-liner I got down to is `(fd as &File).seek(pos)?`. I'm happy to leave several as one-liners such as this one, unless we re-use the `fd` in a couple of places within the same scope, in which case `let mut fd: &File = fd;` seems more natural.